scripts/kf7281/main.py

```diff
@@ -0,0 +1,188 @@
+
+    This is a function with side-effects, that modify the underlying file.
+
+    Args:
+        filename: Path, name of the tox.ini file to be amended
+
+    Returns:
+        true if the file was modified, false if it didn't have the unit env.
+    """
+
+    config = configparser.ConfigParser()
+
+    config.read(filename)
+
+    if (
+            not "testenv:unit" in config.sections() or
+            "coverage xml" in config["testenv:unit"]["commands"]
+    ):
+        return False
+
+    config["testenv:unit"]["commands"] += "\ncoverage xml"
+
+    with open(filename, 'w') as configfile:
+        config.write(configfile)
+
+    return True
+
+def _single_repo_tics(repo_name: str, filename: Path):
+    """Create TIOBE scan workflow for a single charm repository.
+
+    This is a function with side-effects, that creates or overwrites the underlying file.
+
+    Args:
+        repo_name: str, name of the repository / project to be used in TIOBE
+        filename: Path, name of the Github Action file
+    """
+
+    template = Path("./scripts/kf7281/tics-single-repo.yaml.j2")
+
+    env = jinja2.Environment()
+    with open(template, "r") as fid:
+        template = env.from_string(fid.read())
+
+    with open(filename, "w") as fid:
+        fid.write(template.render(
+            project_name=repo_name,
+            tics_auth_token="${{ secrets.TICSAUTHTOKEN }}"
+        ))
+
+
+def _multi_repo_tics(repo_name: str, charms: list[LocalCharmRepo], filename: Path):
+    """Create TIOBE scan workflow for a multi charm repository.
+
+    This is a function with side-effects, that creates or overwrites the underlying file.
+
+    Args:
+        repo_name: str, name of the repository / project to be used in TIOBE
+        charms: list[kfcicli.charms.LocalCharmRepo], list of charms to be included in tiobe scanning
+        filename: Path, name of the Github Action file
+    """
+    template = Path("./scripts/kf7281/tics-multi-repo.yaml.j2")
+
+    env = jinja2.Environment()
+    with open(template, "r") as fid:
+        template = env.from_string(fid.read())
+
+    input_dict = {
+        charm.metadata.name: str(charm.tf_module.parent)
+        for charm in charms
+    }
+
+    with open(filename, "w") as fid:
+        fid.write(template.render(
+            project_name=repo_name,
+            tics_auth_token="${{ secrets.TICSAUTHTOKEN }}",
+            charms=input_dict
+        ))
+
+
+def create_tics_file(repo: Client, charms: list[LocalCharmRepo]):
+    """Create TIOBE scan workflow for a repository (either single-charm or multi-charm).
+
+    This is a function with side-effects, that creates or overwrites the underlying file.
+
+    Args:
+        repo: kfcicli.repository.Client, client instance representing the repository
+        charms: list[kfcicli.charms.LocalCharmRepo], list of charms in the repository
+    """
+
+    filename = repo.base_path / ".github" / "workflows" / "tiobe_scan.yml"
+    if len(charms)==1:
```

Comment:
since we pass `charms_with_unit`, this implicitly expects that no charm `tox.ini` file already contains `coverage xml`. I guess this is true for KF charms, maybe we could have a comment here about this.

Reply:
I can add a check to include the coverage xml if not present. I believe it is sensible

Reply:
actually, the check was already here: https://github.com/deusebio/kubeflow-ci-cli/pull/2/files#diff-749b797cf33e8e4c2b6a74dec5a79fe8a5e104380a770348fa1bb917a4c595e4R44